src/Language/PureScript/CoreFn/Module.hs

```diff
@@ -10,6 +10,9 @@ import Language.PureScript.Types
 -- |
 -- The CoreFn module representation
 --
+-- The json CoreFn representation does not contain type information.  When
```

Comment:
It looks like we don't even use the types in `moduleForeign`. If that's the case, I'd actually rather just get rid of the types completely and lose the type argument.

Reply:
As long as they're kept around in externs, don't drop them there! (Just chiming in as I'm wrestling with an alternative backend *not* targeting a dynamic/untyped language like JS/Py/Erl etc ... from what I can tell I'll have to reconstruct a lot from literals & externs & such as I'm working off CoreImp.) Just a quick reminder to not forget the awesome use-case for alternative backends, some of which may want to keep around as much type info as possible, to minimize usage of boxed-variants (ie what is `object` in Java or `interface{}` in Go, etc) being passed around.

In my fork I experimentally added all `Ann`s to your `--dump-corefn` module, to see what I can get beyond what's in externs. Not *that* much more but last I checked, it did have a bit more detail. A bit exploratory all right now, to write a backend based on --dump-core*foo* outputs rather than right in Haskell in the fork, but it's fun for now. All right I'm done blogging & derailing the thread 😀

Reply:
As far as I can tell, we don't have any type information in the generated JSON right now.